provider/defaultingress/ingress.go

```diff
@@ -389,10 +392,18 @@ func (r *DefaultIngressResource) updateIngress(ctx context.Context, state, plan
 		ingressBuilder := getDefaultIngressBuilder(ctx, plan)
 
 		if !reflect.DeepEqual(state.ClusterRoutesHostname, plan.ClusterRoutesHostname) {
-			ingressBuilder.ClusterRoutesHostname(plan.ClusterRoutesHostname.ValueString())
+			value := ""
+			if !plan.ClusterRoutesHostname.IsNull() {
```

Comment:
What will happen if `state.ClusterRoutesHostname="stringA" && plan.ClusterRoutesHostname = nil` ?
I think we will not update the backend in that case

Reply:
Not sure if I follow @bardielle . If they are different we do want to update to what the plan value is, as it's the new value. If the plan is nil means we want to remove the setting, which is equivalent to supplying '' (empty) to the api

Reply:
The purpose of not allowing empty string in the state is to avoid from having two kinds of valid empty values ("" and null), but we can continue use the null.
When user sets null, its means he wants to delete the existing value, which is legal action.
Having that, null is the only empty value which can be passed to the API and return from it
I agree with @bardielle that we shouldn't ignore this null value and this should resulted by removing the existing value.

Reply:
If I'm not mistaken null is not accepted by the API though. The way to remove the existing value would be to remove the attribute from the TF plan :thinking:

Reply:
Can you test it manually? 
I mean update the attribute `ClusterRoutesHostname` to be null and use the ocm API query to validate it was removed

Reply:
Yes, just finished. Setting to either null or removing the attributes works as expected. Setting the attribute to empty fails the plan step